infra/modules/service/main.tf

```
@@ -29,7 +29,7 @@ locals {
   environment_variables = concat(
     local.base_environment_variables,
     local.db_environment_variables,
-    [{ name : "DRAFTS_S3_BUCKET_ARN", value : aws_s3_bucket.draft_documents.arn }],
+    [{ name : "DRAFTS_S3_BUCKET_ARN", value : one(aws_s3_bucket.draft_documents[*].arn) }],
```

```suggestion
    var.enable_drafts_bucket != null ? [{ name : "DRAFTS_S3_BUCKET_ARN", value : aws_s3_bucket.draft_documents[0].arn }] : [],
```

^ needs to be a ternary that resolves to empty with the input condition is null

---

Actually, flip this ^ that one around...

```suggestion
    var.enable_drafts_bucket == null ? [] : [{ name : "DRAFTS_S3_BUCKET_ARN", value : aws_s3_bucket.draft_documents[0].arn }],
```